src/briefcase/integrations/android_sdk.py

```diff
@@ -1519,4 +1514,12 @@ def kill(self):
         try:
             self.run("emu", "kill")
         except subprocess.CalledProcessError as e:
-            raise BriefcaseCommandError("Error starting ADB logcat.") from e
+            raise BriefcaseCommandError("Error stopping the Android emulator.") from e
+
+    def datetime(self) -> datetime:
+        """Obtain the device's current date/time."""
+        try:
+            # request datetime in seconds since Unix epoch
+            return datetime.fromtimestamp(int(self.run("shell", "date", "+%s")))
```

Comment:
Just confirming: is this using UTC epoch or local timezone epoch?

Reply:
hmm...so, if `datetime.fromtimestamp()` isn't given a timezone, then it returns a naive date/time for the platform's date/time configuration.

In practice, this assumes the device and the host machine are configured for the timezone....and daylight saving...and whatever else....

Reply:
I suppose this requirement could be avoided if a specific format is requested from `date` on the device and an equivalent format is specified for `datetime.strptime()`

Reply:
Look, we can all agree that time was invented by the devil to punish programmers. :-)

We could definitely go to great lengths to ensure timezone compliant round trips - but I'm not sure that's actually required. We just need to make sure that the time that comes back from `adb shell date` is in the same timezone as the log file filter. If that's true with `fromtimestamp()`, then I'm 99% happy; the remaining 1% is to document the new `datetime()` endpoint to make sure nobody tries to use it for anything else in future.

Reply:
> Look, we can all agree that time was invented by the devil to punish programmers. :-)

If that isn't the truth....

> We could definitely go to great lengths to ensure timezone compliant round trips - but I'm not sure that's actually required. We just need to make sure that the time that comes back from `adb shell date` is in the same timezone as the log file filter. If that's true with `fromtimestamp()`, then I'm 99% happy; the remaining 1% is to document the new `datetime()` endpoint to make sure nobody tries to use it for anything else in future.

Well...I guess as long as 99% of the time users' devices have date/time configurations that match their machines that are running Briefcase....then we're good here :/

This is why I had to commit https://github.com/beeware/briefcase/pull/1356/commits/c2e4b6b2c3c222b1a35eedbbb19e0ddb98fc87bf after pushing; I discovered that GitHub's CI runners are configured for UTC (as opposed to my EDT) and were returning a different date/time for the same value returned from `date` on the device.

I could play around with explicit date/time formats tomorrow to see what this takes.

Reply:
[ed9c234](https://github.com/beeware/briefcase/pull/1356/commits/ed9c234f0e1a8ac06237a44d1aad41f03be5524c) will ensure this logging date/time is local to the device.